jmxtrans-cluster/src/main/java/com/googlecode/jmxtrans/cluster/ClusterServiceModule.java

```diff
@@ -0,0 +1,47 @@
+package com.googlecode.jmxtrans.cluster;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import org.apache.commons.configuration.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Created by kulcsart on 5/14/2016.
+ *
+ */
+public class ClusterServiceModule extends AbstractModule {
+    private static final Logger log = LoggerFactory.getLogger(ClusterServiceModule.class);
+    Class implClass;
+    Configuration configuration;
+
+    public ClusterServiceModule(Configuration configuration){
+        this.configuration = configuration;
+
+        checkNotNull(configuration);
+        String className = configuration.getString("provider.classname");
+        checkNotNull(className);
+
+        implClass = getImplClassForName(className);
```

Comment:
Do we really need to have this level of flexibility here? Can't we have just a flag and bind the correct implementation? Do we even need to have multiple possible implementations?